### PR TITLE
Manual mode implementation

### LINKED ISSUE #779: Rewrite TimerComponent to simulate the top row of manualmode
screen shot 2015-06-16 at 11 54 42

### PR DESCRIPTION
**Pull request by taavirehemagi**
Closes #779 #780 #781

### CODE REVIEW

**Comment by monday8am** [comment]
@taavirehemagi the effect - animation not correspond with the Material Design style and don't have enough quality to be an alternative. Besides the code is buggy when you test it.

**Comment by taavirehemagi** [comment]
animation
@monday8am  What do you think of this kind of transition?

**Comment by masterrr** [comment]
Just tested the transition! In general: superb! :+1: 

I noticed one bug:
Launch -> Click "Add Project" / on some TE -> Return by clicking the back arrow (OnCreate exception)

**Comment by masterrr** [comment]
"STARTSTOP" is a mock for a FAB?

**Comment by monday8am** [comment]
@taavirehemagi : the animation is nice! the general aspect is ok, but the code is still buggy. 
There are some movements that show a grey box at bottom, and if press a TE and go back the app crash.

File: Joey/UI/Components/TimerComponent.cs [hunk 1]
```diff
@@ -178,123 +186,63 @@ private void Rebind ()
...
+            DescriptionTextView.Visibility = ViewStates.Visible;
+            DurationTextView.Gravity = GravityFlags.Center;
 
-                // Schedule next rebind:
-                handler.RemoveCallbacks (Rebind);
-                handler.PostDelayed (Rebind, 1000 - duration.Milliseconds);
-            } else {
-                DurationTextView.Visibility = ViewStates.Gone;
+            if (CompactView) {
+                ProjectTextView.Text = currentEntry.Project != null ? currentEntry.Project.Name : "(no project)";
```
Comment (masterrr) [inline Joey/UI/Components/TimerComponent.cs]: We have strings for that
    <string name="RecentTimeEntryNoProject">(no project)</string>

File: Joey/UI/Views/SnappyLayout.cs [hunk 2]
```diff
@@ -68,10 +76,17 @@ private void ForEachChild (Action<int, View> act)
         private void UpdateChildrenTranslationY ()
         {
             translateY = Math.Min (0, Math.Max (maxTranslateY, translateY));
-
+            int childNumber = 0;
             ForEachChild (child => {
-                child.TranslationY = translateY;
+                if (! (child.Top == 0 && lockFirstChild)) {
```
Comment (masterrr) [inline Joey/UI/Views/SnappyLayout.cs]: extra space :stuck_out_tongue:

File: Joey/UI/Activities/MainDrawerActivity.cs [hunk 3]
```diff
@@ -134,6 +138,45 @@ protected override void OnCreateActivity (Bundle state)
...
+        {
+            switch (toolbarMode) {
+            case MainDrawerActivity.ToolbarModes.DurationOnly:
+                Timer.Hide = false;
+                Timer.CompactView = false;
+                SupportActionBar.SetBackgroundDrawable (null);
+                SupportActionBar.SetDisplayShowTitleEnabled (false);
+                MainToolbarShadow.Visibility = ViewStates.Gone;
+                break;
+            case MainDrawerActivity.ToolbarModes.Compact:
```
Comment (masterrr) [inline Joey/UI/Activities/MainDrawerActivity.cs]: DurationOnly / Compact modes share the same values for:
* Timer.Hide
* SetBackgroundDrawable
* SetDisplayShowTitleEnabled
* MainToolbarShadow.Visibility

Maybe assign these outside the switch, make an if for the Normal mode and then
Timer.CompactView = toolbarMode == MainDrawerActivity.ToolbarModes.Compact;

Or just make a multiple-case, like 
case 1: case 2: 
do() 
break;

**Comment by masterrr** [comment]
Just tested the last revision

> if press a TE and go back the app crash
Still same goes for me too

The animation is still neat for me :) Smooth and cool

**Comment by monday8am** [comment]
@taavirehemagi  this is a list of what I found:

Android 5
- Double shadow at top.
- Remove background Tint (this attribute makes the app crash)

General
- Remove focus while scrolling and returns to forms when it appears.
- Middle position scroll and form doesn't snap to top or bottom (low priority)
- Is hard to know how to edit the time value (Heidi)
- Fab align.
- Wrong align when Fab moves (related with previous point)
- Drawer button align.

Which is the best way for you to work? We could create individual issues and mark them as planed to be fixed in the next two weeks.

**Comment by taavirehemagi** [comment]
Outdated

**Pull request closed without merging**